examples/sample-operator/pkg/controller/sampleconfig/cr-manager.go

```diff
@@ -0,0 +1,114 @@
+package sampleconfig
+
+import (
+	"github.com/kubevirt/controller-lifecycle-operator-sdk/examples/sample-operator/pkg/apis/sample/v1alpha1"
+	sdkapi "github.com/kubevirt/controller-lifecycle-operator-sdk/pkg/sdk/api"
+	"github.com/kubevirt/controller-lifecycle-operator-sdk/pkg/sdk/resources"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	httpServerDeploymentName = "http-server-deployment"
+	httpServerName           = "http-server"
+	deploymentMatchKey       = "sample-operator.kubevirt.io"
+	httpServerPort           = 8081
+)
+
+const (
+	commonLabel   = "test.common"
+	operatorLabel = "test.operator"
+)
+
+var (
+	commonLabels    = map[string]string{commonLabel: ""}
+	operatorLabels  = map[string]string{operatorLabel: ""}
+	resourceBuilder = resources.NewResourceBuilder(commonLabels, operatorLabels)
+)
+
+// SampleConfigCrManager provides test CR management functionality
+type CrManager struct {
+	operatorArgs *OperatorArgs
+}
+
+// IsCreating checks whether creation of the managed resources will be executed
+func (m *CrManager) IsCreating(cr controllerutil.Object) (bool, error) {
+	config := cr.(*v1alpha1.SampleConfig)
+	return config.Status.Conditions == nil || len(config.Status.Conditions) == 0, nil
+}
+
+// Create creates empty CR
+func (m *CrManager) Create() controllerutil.Object {
+	return new(v1alpha1.SampleConfig)
+}
+
+// Status extracts status from the cr
+func (m *CrManager) Status(cr runtime.Object) *sdkapi.Status {
+	return &cr.(*v1alpha1.SampleConfig).Status.Status
+}
+
+// GetAllResources provides all resources managed by the cr
+func (m *CrManager) GetAllResources(cr runtime.Object) ([]runtime.Object, error) {
+	namespace := m.operatorArgs.Namespace
+
+	serviceAccount := resourceBuilder.CreateServiceAccount(httpServerName)
+	serviceAccount.Namespace = namespace
+
+	role := resourceBuilder.CreateRole(httpServerName, []rbacv1.PolicyRule{})
+	role.Namespace = namespace
+
+	roleBinding := resourceBuilder.CreateRoleBinding(httpServerName, httpServerName, httpServerName, namespace)
+	roleBinding.Namespace = namespace
+
+	httpDeployment := m.createHTTPServerDeployment()
+	httpService := m.createHTTPServerService()
```

Comment:
We could add http service definition here.

Reply:
:+1: I will expose the HTTP server port on the node in the server service definition.